core/src/test/java/com/opensymphony/xwork2/util/MyBeanActionTest.java

```diff
@@ -89,6 +107,21 @@ public void testIndexedMap() {
 
             assertEquals("1234567890", proxy.getInvocation().getStack().findValue("beanMap.get(1234567890L).id").toString());
             assertEquals("1234567891", proxy.getInvocation().getStack().findValue("beanMap.get(1234567891L).id").toString());
+
+            assertEquals(2, Integer.parseInt(proxy.getInvocation().getStack().findValue("annotatedBeanMap.size").toString()));
+
+            assertEquals(true, action.getAnnotatedBeanMap().containsKey(1234567890L));
+            assertEquals(true, action.getAnnotatedBeanMap().containsKey(1234567891L));
```

Comment:
I think it would be good to check values of these keys.

Reply:
I see they are already checked in it's following lines using `findValue` :)

Reply:
hm... `findValue()` performs a scan over the stack so it can find other values than expected but I think in this case it's ok.